app.js

```diff
@@ -0,0 +1,299 @@
+      });
+
+      const dockerEnvironmentalVariables = volumeDefinitions.reduce(
+        (acc, cur) => ({
+          ...acc,
+          [cur.mountPoint.name]: cur.mountPoint.value,
+        }),
+        {},
+      );
+
+      const shellEnvironmentalVariables = volumeDefinitions.reduce(
+        (acc, cur) => ({
+          ...acc,
+          [cur.path.name]: cur.path.value,
+        }),
+        dockerEnvironmentalVariables,
+      );
+
+      const helmCommand = `\
+uninstall \
+${chartName} \
+${installationParameters.join(" ")} \
+${authenticationParameters.join(" ")}`;
+
+      const command = docker.buildDockerCommand({
+        command: helmCommand,
+        image: HELM_IMAGE_NAME,
+        environmentVariables: dockerEnvironmentalVariables,
+        volumeDefinitionsArray: volumeDefinitions,
+      });
+
+      result = await exec(command, {
+        env: shellEnvironmentalVariables,
+      });
+    },
+  );
+
+  const { stdout, stderr } = result;
+
+  if (!stdout && stderr) {
+    throw new Error(stderr);
+  }
+
+  return stdout;
+}
+
+async function runCommand(pluginParameters) {
+  const {
+    kubeCertificate,
+    cliCommand,
+  } = pluginParameters;
+
+  let result;
+
+  await helpers.temporaryFileSentinel(
+    [kubeCertificate],
+    async (certificateFilePath) => {
+      const [certificatePath, certificateFileName] = splitDirectory(certificateFilePath);
+      const certificateVolumeDefinition = docker.createVolumeDefinition(certificatePath);
+      const volumeDefinitions = [
+        certificateVolumeDefinition,
+      ];
+
+      const authenticationParameters = generateAuthenticationParameters({
+        certificateFilePath: `$${certificateVolumeDefinition.mountPoint.name}/${certificateFileName}`,
+        kubeToken: pluginParameters.kubeToken,
+        kubeApiServer: pluginParameters.kubeApiServer,
+        kubeUser: pluginParameters.kubeUser,
+      });
+
+      const dockerEnvironmentalVariables = volumeDefinitions.reduce(
+        (acc, cur) => ({
+          ...acc,
+          [cur.mountPoint.name]: cur.mountPoint.value,
+        }),
+        {},
+      );
+
+      const shellEnvironmentalVariables = volumeDefinitions.reduce(
+        (acc, cur) => ({
+          ...acc,
+          [cur.path.name]: cur.path.value,
+        }),
+        dockerEnvironmentalVariables,
+      );
+
+      const helmCommand = `\
+${sanitizeCommand(cliCommand)} \
+${authenticationParameters.join(" ")}`;
+
+      const command = docker.buildDockerCommand({
+        command: helmCommand,
+        image: HELM_IMAGE_NAME,
+        environmentVariables: dockerEnvironmentalVariables,
+        volumeDefinitionsArray: volumeDefinitions,
+      });
+
+      result = await exec(command, {
+        env: shellEnvironmentalVariables,
+      });
+    },
+  );
+
+  const { stdout, stderr } = result;
+
+  if (!stdout && stderr) {
+    throw new Error(stderr);
+  }
+
+  return stdout;
+}
+
+function generateAuthenticationParameters(pluginParams) {
```

Comment:
In my opinion, let's export these "helper" functions into a separate file, preferably **helm-cli.js**, just like the core logic.

Reply:
Fixed